.github/workflows/js_tests.yml

```diff
@@ -0,0 +1,272 @@
+name: JS tests master
+
+on:
+  push:
+    branches:
+      - master
+      - 3.x-maintenance
+    paths:
+      - '.github/scripts/*.sh'
+      - '.github/workflows/*.yml'
+      - 'gui/package.json'
+      - 'requirements/*.txt'
+      - 'Makefile'
+      - 'MANIFEST.in'
+      - '**.bat'
+      - '**.jsx?'
+      - '**.sh'
+
+  pull_request:
+    branches:
+      - master
+      - 3.x-maintenance
+    paths:
+      - '.github/scripts/*.sh'
+      - '.github/workflows/*.yml'
+      - 'gui/package.json'
+      - 'requirements/*.txt'
+      - 'MANIFEST.in'
+      - '**.bat'
+      - '**.jsx?'
+      - '**.sh'
+
+jobs:
+  ############################################################################
+  # Close previous builds on the same branch
+  ############################################################################
+  cleanup-previous-runs:
+    name: Cleanup previous builds
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo $GITHUB_EVENT_NAME
+      - uses: rokroskar/workflow-run-cleanup-action@v0.2.2
+        if: github.event_name == 'pull_request'
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+
+  ############################################################################
+  # Lint jobs
+  ############################################################################
+  lint-js:
+    name: Lint JS
+    needs: cleanup-previous-runs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Fetch branches
+        run: git fetch --prune --unshallow
+      - name: Check build skips
+        shell: bash -l {0}
+        run: source .github/scripts/check_skip.sh
+      - name: Cache node
+        if: env.RUN_BUILD == 'true'
```

Comment:
Can this work instead?

https://github.com/srz-zumix/ci-skip/blob/master/.github/workflows/main.yml

Reply:
Not for prs